### PR TITLE
Uploading binaries to pre-release on tag creation

### PR DESCRIPTION
**Pull request by GerardoGR**
I would like to have the binaries compiled for the released version in the release assets section. This PR introduces that and also creates a pre-release on tag creation. This a proposal to solve an issue I created in GitHub #1161 and asked in the matrix channel.

### CODE REVIEW

**Comment by Ornias1993** [comment]
Yes this would be very nice and I could also hook into this when we get #1154 merged. :)

File: .github/workflows/workspace.yml [hunk 1]
```diff
@@ -139,10 +143,66 @@ jobs:
...
+  upload-release-assets:
+    name: Upload release assets
+    if: startsWith(github.ref, 'refs/tags/')
+    needs: [build, prepare-release]
+    strategy:
+      fail-fast: false
+      # TODO: YAML Anchors could reduce the matrix repetition but they are not
+      # supported https://github.community/t/support-for-yaml-anchors/16128
+      matrix: 
+        db-backend: [sqlite, mysql, postgresql]
```
Comment (Ornias1993) [inline .github/workflows/workspace.yml:174]: I think the use of matrix is fine, not super efficient but it isn't running that often anyway :)

File: .github/workflows/workspace.yml [hunk 2]
```diff
@@ -139,10 +143,66 @@ jobs:
...
+        include:
+          - target: x86_64-unknown-linux-gnu
+            os: ubuntu-latest
+            ext:
+          # - target: x86_64-unknown-linux-musl
+          #   os: ubuntu-latest
+          #   ext:
+          # - target: x86_64-apple-darwin
+          #   os: macOS-latest
+          #   ext:
```
Comment (Ornias1993) [inline .github/workflows/workspace.yml:189]: On a later date we should look into packaging this into a nice dmg or something.
But thats a bit offtopic.

File: .github/workflows/workspace.yml [hunk 3]
```diff
@@ -4,6 +4,10 @@ on:
   push:
     paths-ignore:
       - "**.md"
+    branches:
+      - "*"
```
Comment (Ornias1993) [inline .github/workflows/workspace.yml:8]: I don't think branches* is needed with tags*

**Review by Ornias1993 — APPROVED** [review]
Generally fine, one slight footnote.

**Comment by BlackDex** [comment]
The issue i have with this is that what is build isn't a static binary.
You need the correct library versions installed on your system else this will not work, or cause unexpected issues.
This is where the docker containers are great in by providing these exact same libraries as where the binary is build with.

If we want to provide this, i think we should mention which OS was used to build it on.
So, having a musl static build version for any OS would get a clear go from my side, but in my experience this is only going to cause confusion and weird issue we will get reported here.

I don't know how @dani-garcia or @jjlin think about this, but for me a musl version would be ok, dynamic ones no.

**Comment by jjlin** [comment]
I agree that for this to be broadly useful, the binaries would have to be statically linked. Or there would have to be separate builds uploaded for each distro/version being targeted, which could get a bit messy.

**Comment by BlackDex** [comment]
@GerardoGR, if you are able to get static binaries build using gitlab actions i would be happy to say yes to this.
But for now, i think we should close this PR until it is possible to have static builds which would work on almost all Linux distributions.

**Comment by GerardoGR** [comment]
@BlackDex I will have a look on how to build the static binaries. If it is fine for you, I will close this PR in the meanwhile.

**Pull request closed without merging**